miniapp/CMakeLists.txt

```diff
@@ -18,6 +18,7 @@ else()
     add_executable(miniapp.exe ${MINIAPP_SOURCES} ${HEADERS})
 endif()
 
+target_link_libraries(miniapp.exe LINK_PUBLIC ${CMAKE_THREAD_LIBS_INIT} )
```

Comment:
I'll see whether this works throughout the tree, rather than having these target links everywhere